### ISSUE TITLE
Enable github cache for mvn dependencies

### ISSUE THREAD

**Issue by ccoltx** [codecentric/spring-boot-admin#1677]
Decrease build times and improve build reliability, enable the github cache action for mvn dependencies.

https://github.com/actions/cache
https://docs.github.com/en/actions/guides/caching-dependencies-to-speed-up-workflows

**Comment by jonashackt** [comment]
LGTM! :) What is this annoying 

```
Error:  Plugin pl.project13.maven:git-commit-id-plugin:4.0.3 or one of its dependencies could not be resolved: Failed to read artifact descriptor for pl.project13.maven:git-commit-id-plugin:jar:4.0.3: Could not transfer artifact pl.project13.maven:git-commit-id-plugin:pom:4.0.3 from/to central (https://repo.maven.apache.org/maven2): Transfer failed for https://repo.maven.apache.org/maven2/pl/project13/maven/git-commit-id-plugin/4.0.3/git-commit-id-plugin-4.0.3.pom: Connection timed out (Read failed) -> [Help 1]
```

all about? Is there a problem with Maven Central access? Should we upgrade to the latest 4.0.4? I don't really get the error really... since why does this dependency fail and not others if there were a Maven Central bug...?

**Comment by Riggs333** [comment]
The update to version 4.0.4 of the git-commit-id-plugin does not seem to help, see PR #1663.
Same error message during the build. :-(
But the build fails only on one of the four checks. 🤔

**Comment by jonashackt** [comment]
What for do we use the git-commid-id-plugin? Maybe we could get rid of it...?

**Comment by joshiste** [comment]
> What for do we use the git-commid-id-plugin? Maybe we could get rid of it...?

It creates a git.properties file in the sample applications. This information is automatically picked up by spring boot and served on the info endpoint. 
So the plugin is not vital but shows a simple way for an integration to get the git information in SBA

**Comment by ccoltx** [comment]
There is an issue with the `actions/cache@v2` version. The fix is included in `actions/cache@v2.1.4`. See:

https://www.gitmemory.com/issue/actions/toolkit/632/743759333
https://github.com/actions/toolkit/issues/632#issuecomment-810139715
